### PR TITLE
particles rotation support (bug #4883)

### PR DESCRIPTION
**Pull request by mp3butcher**
hard to find test subjects
snow
https://streamable.com/t4xxq
fountain
https://streamable.com/946w0
NB: not compatible with point sprite shading  https://github.com/OpenMW/openmw/pull/2441 ffp only
in any case it doesnt solve https://gitlab.com/OpenMW/openmw/issues/4783

### CODE REVIEW

File: components/nifosg/nifloader.cpp [hunk 1]
```diff
@@ -811,7 +811,12 @@ namespace NifOsg
                 }
                 else if (affectors->recType == Nif::RC_NiParticleRotation)
                 {
-                    // unused
+                    const Nif::NiParticleRotation *cl = static_cast<const Nif::NiParticleRotation*>(affectors.getPtr());
+                    /*OSG_WARN << "handling RC_NiParticleRotation modifier " << affectors->recName << " in " << mFilename;
+                    OSG_WARN << cl->mRandomInitialAxis <<std::endl;
+                    OSG_WARN << cl->mRotationSpeed <<std::endl;
```
Comment (Capostrophic) [inline components/nifosg/nifloader.cpp]: Please remove obvious debug remnants.

File: components/nifosg/particle.cpp [hunk 2]
```diff
@@ -153,6 +153,36 @@ void ParticleColorAffector::operate(osgParticle::Particle* particle, double /* d
...
+    : osgParticle::Operator(copy, copyop)
+{
+    *this = copy;
+}
+
+void ParticleRotationAffector::operate(osgParticle::Particle* particle, double /* dt */)
+{
+    if(particle->getAge() == 0)
+    {
+        osg::Vec3 initialAxis = osg::Vec3f(Misc::Rng::rollClosedProbability() * 2.0 * osg::PI,
```
Comment (Capostrophic) [inline components/nifosg/particle.cpp]: Initial axis aren't supposed to be random if the relevant flag is not set.

File: components/nifosg/nifloader.cpp [hunk 3]
```diff
@@ -811,7 +811,12 @@ namespace NifOsg
                 }
                 else if (affectors->recType == Nif::RC_NiParticleRotation)
                 {
-                    // unused
```
Comment (Capostrophic) [inline components/nifosg/nifloader.cpp]: Would really be nice to get a verification of this before we suddenly start to do something vanilla doesn't.

File: components/nifosg/particle.cpp (hunk 2 above)
Comment (mp3butcher) [reply components/nifosg/particle.cpp]: arf forgot a test ...

File: components/nifosg/nifloader.cpp (hunk 3 above)
Comment (mp3butcher) [reply components/nifosg/nifloader.cpp]: noone anwers me on discord....

**Comment by Capostrophic** [comment]
Random initial axis must replace initial axis completely instead of multiplying them.

**Comment by mp3butcher** [comment]
..initial axis is of 0 0 0 most of the time

**Comment by mp3butcher** [comment]
> Random initial axis must replace initial axis completely instead of multiplying them.

..no..only rotation around axis must be randomized

File: components/nifosg/nifloader.cpp (hunk 3 above)
Comment (mp3butcher) [reply components/nifosg/nifloader.cpp]: @EquiNox seems to say rotations aren't part of vanilla...

**Comment by mp3butcher** [comment]
After checking, rotations are there but not used in vanilla...so i close this pr

**Comment by mp3butcher** [comment]
again for testing reopening..please don't bother

**Pull request closed without merging**